### DISCUSSION TITLE
Softflow Integration on pfSense was pretty quick

### DISCUSSION THREAD

**Discussion by nonanonymousanon** [pfelk/pfelk#304] [Show and tell]
For those interested net flow for logstash can be added with one line of code (though it is deprecated). I haven't tried to do it through filebeat yet so if someone has that prepared please feel free to share it. 

`bin/logstash --modules netflow --setup -M netflow.var.input.udp.port=2055 -M netflow.var.elasticsearch.hosts="localhost:9200" -M netflow.var.elasticsearch.ssl.enabled=false -M netflow.var.kibana.host="localhost:5601" -M netflow.var.kibana.scheme=http -M netflow.var.kibana.ssl.enabled=false -M netflow.var.kibana.ssl.verification_mode=disable`

For persistence I had to add the below module to the logstash.yml

modules:
    - name: netflow
       var.input.udp.port: 2055
       var.elasticsearch.hosts: "localhost:9200
       var.elasticsearch.ssl.enabled: false
       var.kibana.host: "localhost:5601"
       var.kibana.scheme: http
       var.kibana.ssl.enabled: false
       var.kibana.ssl.verification_mode: disabled

Configuration in pfsense looks like this. I had trouble with net flow version 9 but 5 seems to be working more properly. 

![Screen Shot 2021-05-12 at 7 13 45 PM](https://user-images.githubusercontent.com/68710948/118068676-9e0b4f00-b357-11eb-9c01-4f29aef2fa16.png)

Allows net flow tracking on any interface on pfsense and gives a few premade dashboards in kibana. Feedback encouraged.

Based off this guide: [https://hilo21.com/th/threat-hunting-lab-part-ii-sending-pfsense-netflow-data-to-elastic-stack/(url)

**Comment by a3ilson** [comment]
@ckamerin - thanks! Unfortunately, Logstash Netflow Module has been deprecated since v7.4.0

Reference:
https://www.elastic.co/guide/en/logstash/current/netflow-module.html
